app/scripts/fabric-extensions/rescale-2-resize.js

```diff
@@ -40,20 +40,15 @@ var duringResize = {
     this.line(s);
   },
   path: function (s) {
+    var offsetX = s.pathOffset.x;
+    var offsetY = s.pathOffset.y;
     for (var i = 0; i < s.path.length; i++) {
-      s.path[i][1] *= s.scaleX;
-      s.path[i][2] *= s.scaleY;
-      s.path[i][3] *= s.scaleX;
-      s.path[i][4] *= s.scaleY;
+      s.path[i][1] = (s.path[i][1] - offsetX) * s.scaleX + offsetX;
+      s.path[i][2] = (s.path[i][2] - offsetY) * s.scaleY + offsetY;
+      s.path[i][3] = (s.path[i][3] - offsetX) * s.scaleX + offsetX;
+      s.path[i][4] = (s.path[i][4] - offsetY) * s.scaleY + offsetY;
```

Comment:
The idea is to scale path coordinates without offset. Offset remains unchanged. That's why I first subtract and then add the same value. 

You can test this behaviour if you open this page:
http://concord-consortium.github.io/drawing-tool/examples/
select free hand drawing tool (4th icon), draw anything, then select it (1st icon) and try to resize.